Add documentation changes for shallow snapshot v2 (#8207)
opensearch-project/documentation-website: _api-reference/snapshots/create-snapshot.md

| failures | array | Failures, if any, that occured during snapshot creation. |
| shards | object | Total number of shards created along with number of successful and failed shards. |
| state | string | Snapshot status. Possible values: `IN_PROGRESS`, `SUCCESS`, `FAILED`, `PARTIAL`. |

> | remote_store_index_shallow_copy | Boolean | Whether the snapshot of the remote store indexes is captured as a shallow copy. Default is `false`. |

Reviewer: "snapshots of the remote store indexes are"?

> | remote_store_index_shallow_copy | Boolean | Whether the snapshot of the remote store indexes is captured as a shallow copy. Default is `false`. |

Reviewer 2: ```suggestion
| remote_store_index_shallow_copy | Boolean | Whether the snapshots of the remote store indexes is captured as a shallow copy. Default is `false`. |
```

> | pinned_timestamp | long      | Timestamp (in milliseconds) that is pinned by the snapshot for implicit locking of remote store files referenced by the snapshot. |

Reviewer 2: ```suggestion
| pinned_timestamp | long      | A timestamp (in milliseconds) pinned by the snapshot for the implicit locking of remote store files referenced by the snapshot. |
```
